Feature/collapse location cards

schamp opened the pull request:
Update based on designs: https://www.figma.com/file/JGqemgZwlG9e5DQbZpdKgy/Redesign?node-id=2269%3A88

After:
Desktop:
![image](https://user-images.githubusercontent.com/351548/118424567-4e40c680-b67c-11eb-9960-961776a93175.png)
375x800: (note search bar is full width)
![image](https://user-images.githubusercontent.com/351548/118424700-8942fa00-b67c-11eb-94df-868413e10d72.png)
320x568: (note fixed search bar -- doesn't extend beyond screen)
![image](https://user-images.githubusercontent.com/351548/118424755-a4156e80-b67c-11eb-9f1f-e4371bbe9ec1.png)
320x568 (scrolled):
![image](https://user-images.githubusercontent.com/351548/118424796-b7283e80-b67c-11eb-9164-92589fedcd1a.png)

schamp commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Collapse location cards on homepage](https://trello.com/c/10mKstRR/1234-collapse-location-cards-on-homepage)

schamp commented:
If you have any guidance about a better way to organize the styles or apply properties to achieve the desired result, let me have it!  I got this working, but my CSS game isn't as strong as I'd like and there may well be better ways to accomplish this.

joshziman commented:
Looking good @schamp! I wasn't able to load the cards, perhaps for the reason you mentioned in Trello. Judging from the screenshots, on mobile, the only thing I'd request changed is to have the card "full bleed" to the edge of the screen, i.e. there wouldn't be a gap right around "VA" in the screenshot.

schamp commented:
Cool, thanks, I'll see what I can do about that.

mimecuvalo commented on src/screens/HomePage/HomePage.style.tsx:
```diff
@@ -35,30 +29,19 @@ export const Section = styled.div`
-    }
-  }
+  flex-direction: row;
+  gap: 0.75rem;
```
TIL about `gap` - neat!

chasulin replied:
ooooo same!!

mimecuvalo commented on src/components/RegionItem/RegionItem.style.tsx:
```diff
@@ -51,9 +65,13 @@ export const SharedWrapperStyles = css`
   border-radius: 4px;
+
+  @media (min-width: ${materialSMBreakpoint}) {
+    width: 296px;
```
nit: might be good to make this a constant with above

schamp replied:
good idea, thanks

mimecuvalo reviewed (approved):
LGTM with caveat that i'm still ramping on this code
(and also, hopefully we'll have constants long-term in our other codebases vs. hard-coded numbers but I'm not worried for the CAN site in this regard)

chasulin commented on src/assets/theme/searchbar.ts:
```diff
@@ -16,7 +16,7 @@ export interface SearchbarTheme {
 export const searchbar: SearchbarTheme = {
   mobileClosedWidth: `${mobileWidth}px`,
-  mobileClosedPadding: '0.75rem 0.5rem 1.5rem',
+  mobileClosedPadding: '0.75rem 0.5rem 1rem',
```
lets keep this `1.5rem` to keep it close to [the mocks](https://www.figma.com/file/JGqemgZwlG9e5DQbZpdKgy/Redesign?node-id=1476%3A143)

chasulin commented on src/components/Search/SearchAutocomplete/SearchAutocomplete.style.tsx:
```diff
@@ -55,12 +55,14 @@ const MobileWrapperClosed = css`
 export const Wrapper = styled.div<{ $isOpen: boolean }>`
   ${({ $isOpen }) => ($isOpen ? MobileWrapperOpened : MobileWrapperClosed)};
+  width: 100%;
+  max-width: ${mobileWidth}px;
```
this^ results in the menu not being full bleed on mobile:

<img width="377" alt="Screen Shot 2021-05-17 at 6 39 36 PM" src="https://user-images.githubusercontent.com/44076375/118564948-4eee6100-b73f-11eb-8890-f5a1d715f47d.png">

schamp replied:
nice catch, thanks.  removing both is fine, I didn't need them after all, so long as I change the 'closed' one in the theme from '350px' to '100%';

chasulin commented on src/components/Search/SearchAutocomplete/SearchAutocomplete.style.tsx:
```diff
@@ -55,12 +55,14 @@ const MobileWrapperClosed = css`
   padding: ${props => props.theme.searchbar.mobileClosedPadding};
-  margin-left: ${props => props.theme.searchbar.mobileClosedMarginLeft};
   max-width: ${props => props.theme.searchbar.mobileClosedMaxWidth};
+  margin: auto;
```
oo any reason to edit these margins? nixing the margin-left and adding margin:auto throws off the spacing and width of the searchbar slightly

prod (search bar smaller and set to the right)-
<img width="377" alt="Screen Shot 2021-05-17 at 6 49 11 PM" src="https://user-images.githubusercontent.com/44076375/118565779-c96bb080-b740-11eb-891b-e447504f5909.png">

feature branch-
<img width="378" alt="Screen Shot 2021-05-17 at 6 49 20 PM" src="https://user-images.githubusercontent.com/44076375/118565766-c40e6600-b740-11eb-896f-6fdb33265b68.png">

schamp replied:
reverting this seemed to work, thanks for the callout.

schamp commented:
fixed the padding to match the design more closely (was too small)
![image](https://user-images.githubusercontent.com/351548/118583513-0c7e5180-b74a-11eb-91b4-9f4758d092eb.png)

fixed full bleed on mobile search open: 
![image](https://user-images.githubusercontent.com/351548/118583578-24ee6c00-b74a-11eb-8cbd-6faf5b78d9fc.png)

looks good small too:
![image](https://user-images.githubusercontent.com/351548/118583621-36377880-b74a-11eb-98e3-90ef5e6f0056.png)

search on 320px screen:
![image](https://user-images.githubusercontent.com/351548/118583671-494a4880-b74a-11eb-993f-b5b583df49bf.png)

search on 375px screen (back to original on prod):
![image](https://user-images.githubusercontent.com/351548/118583713-5830fb00-b74a-11eb-8eaf-b4b74e23c434.png)

region cards now full bleed on mobile:  
![image](https://user-images.githubusercontent.com/351548/118583769-71d24280-b74a-11eb-8d89-a6e36093d6b9.png)

I think that's everything, @chasulin do you mind confirming and stamping when you're happy with it?  then I'll merge.

pnavarrc reviewed (approved):
Looks good, 🚢 !
